controllers/clustertopology_controller_compute.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// getNestedRef returns the ref value of a nested field.
+// NOTE: This function assumes the reference either exists in the CRD, and in that case the value is required,
+// or it does not exists in the CRD.
+func getNestedRef(obj *unstructured.Unstructured, fields ...string) *corev1.ObjectReference {
+	if obj == nil {
+		return nil
+	}
+	ref, ok, err := unstructured.NestedMap(obj.UnstructuredContent(), fields...)
+	if !ok || err != nil {
+		return nil
```

Comment:
We don't do anything with err if err != nil. Would it be worth adding a comment here explaining why we drop that?

Reply:
It is already described in the func description that this field is optional (and in that case we return nil)